lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,1683 @@
+        .with_manufacturers(['Microtech'])
+        .with_serialNumber(cfg.deviceInfo.serialNumber)
+        .with_deviceTime(sundial.formatDeviceTime(new Date('2020-02-28 10:00:00').valueOf()))
+        .with_time(
+          sundial.applyTimezone(new Date(), cfg.timezone).toISOString()
+        )
+        .with_timezoneOffset(0)
+        .with_conversionOffset(0)
+        .done();
+
+      return postsettings;
+    },
+
+    buildBlood(BloodGlucoses) {
+      let res = [];
+      for (let blood of BloodGlucoses) {
+        if (new Date(blood.deviceTime).valueOf() < cfg.lastUpload) {
+          continue;
+        }
+        const recordBuilder = cfg.builder
+          .makeSMBG()
+          .with_value(parseFloat(blood.BloodGlucose))
+          .with_units('mg/dL') // values are always in 'mg/dL'
+          .with_deviceTime(
+            sundial.formatDeviceTime(new Date(blood.deviceTime).valueOf())
+          )
+          .set('index', blood.recordId);
+
+        cfg.tzoUtil.fillInUTCInfo(
+          recordBuilder,
+          new Date(blood.deviceTime).valueOf()
+        );
+        const postRecord = recordBuilder.done();
+        delete postRecord.index;
+        res.push(postRecord);
+      }
+      return res;
+    },
+
+    buildBasal(BasalRates) {
+      let res = [];
+      //date sort desc
+      BasalRates.sort(function(a, b) {
+        return a.deviceTime < b.deviceTime ? -1 : 1;
+      });
+      //today
+      // BasalRates = BasalRates.filter((a)=>{ return a.deviceTime > '2020-07-20 23:59:59' && a.deviceTime < '2020-07-22 00:00:00'})
+      for (let i = 0; i < BasalRates.length; i++) {
+        if(BasalRates[i].recordId == 515){
+          console.log(BasalRates[i]);
+        }
+        let currDu = new Date(BasalRates[i].deviceTime).valueOf();
+        let nextDu = BasalRates[i + 1]
+          ? new Date(BasalRates[i + 1].deviceTime).valueOf()
+          : currDu + 1000;
+        if (nextDu - currDu < 0) {
+          debug('error-basal', BasalRates[i]);
+        }
+        if (i == BasalRates.length - 1) {
+          break;
+        };
+        let currentDur = (nextDu - currDu) < 604800000 ? (nextDu - currDu) : 604799999;
+        let basalBuilder = cfg.builder
+          .makeScheduledBasal()
+          .with_scheduleName('basal_1')
+          .with_deviceTime(
+            sundial.formatDeviceTime(
+              new Date(BasalRates[i].deviceTime).valueOf()
+            )
+          )
+          // .with_rate(1)
+          .with_rate(BasalRates[i].BasalRate)
+          .with_duration(currentDur)
+          .set('index', BasalRates[i].recordId);
+        cfg.tzoUtil.fillInUTCInfo(
+          basalBuilder,
+          new Date(BasalRates[i].deviceTime).valueOf()
+        );
+        const postRecord = basalBuilder.done();
+        if (res[i - 1]) {
+          let preRes = JSON.stringify(res[i - 1]);
+          preRes = JSON.parse(preRes);
+          delete preRes.previous;
+          postRecord.previous = preRes;
+        }
```

Comment:
`.previous` has been deprecated, so this can be removed.

Reply:
> `.previous` has been deprecated, so this can be removed.

@gniezen , OK, I will submit these changes as soon as possible